### PR TITLE
Vb/test multiple sdk versions plt 1320

### PR DESCRIPTION
**Pull request by vbrodsky**
# Description

We now pull 4 latest sdk releases and run testing for all PRs to develop.  Oldest release is assigned to the oldest python version and current branch code runs off the latest python version.

Lints remains as is - all python versions for current branch

Prod remains as is - only latest release on all python versions

Screenshot 2024-09-06 at 12 31 45 PM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document change (fix typo or modifying any markdown files, code comments or anything in the examples folder only)

## All Submissions

- [ ] Have you followed the guidelines in our Contributing document?
- [ ] Have you provided a description?
- [ ] Are your changes properly formatted?

## New Feature Submissions

- [ ] Does your submission pass tests?
- [ ] Have you added thorough tests for your new feature?
- [ ] Have you commented your code, particularly in hard-to-understand areas?
- [ ] Have you added a Docstring?

## Changes to Core Features

- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully run tests with your changes locally?
- [ ] Have you updated any code comments, as applicable?

### CODE REVIEW

File: .github/workflows/python-package-develop.yml [hunk 1]
```diff
@@ -38,15 +61,19 @@ jobs:
           - python-version: 3.8
             api-key: STAGING_LABELBOX_API_KEY_2
             da-test-key: DA_GCP_LABELBOX_API_KEY
+            sdk-version: ${{ fromJson(needs.get_sdk_versions.outputs.sdk_versions)[3] }}
```
Comment (adrian-chang) [inline .github/workflows/python-package-develop.yml:64]: Do we not want to test each version against each python version? This tests 1 version against 1 python version

File: .github/workflows/python-package-develop.yml [hunk 2]
```diff
@@ -28,8 +28,31 @@ jobs:
           filters: |
             labelbox:
               - 'libs/labelbox/**'
+  get_sdk_versions:
+    runs-on: ubuntu-latest
+    outputs:
+      sdk_versions: ${{ steps.get_sdk_versions.outputs.sdk_versions }}
+    steps:
+          - name: Checkout repository
```
Comment (adrian-chang) [inline .github/workflows/python-package-develop.yml]: Fixup the formatting here to two spaces

**Review by adrian-chang — CHANGES_REQUESTED** [review]

File: .github/workflows/python-package-develop.yml (hunk 1 above)
Comment (vbrodsky) [reply .github/workflows/python-package-develop.yml:64]: So this is a good question. If we do 5 sdk versions vs 5 python versions that is 25 pipelines. I do not think we can run it on stage.
In my experience, if we have issues for older sdk versions that would be due to api changes
I wanted to run this test for every PR to detect early if there are any backward-compatibility issues in the api

File: .github/workflows/python-package-develop.yml (hunk 2 above)
Comment (vbrodsky) [reply .github/workflows/python-package-develop.yml]: @adrian-chang fixed

**Review by adrian-chang — APPROVED** [review]

**Pull request merged**